iothub/service/tests/Microsoft.Azure.Devices.Tests.csproj

```diff
@@ -1,9 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">netcoreapp2.1</TargetFrameworks>
```

Comment:
unit tests on net451 and net472?

Reply:
I'm getting a build error that I can't figure out atm, so I held off on that part thinking the rest of this should go in asap.

Reply:
Since this PR is ready to go in now, I'll create a branch off of this and submit a PR for the CodeMaid setting.